tests/app/dao/test_service_permissions_dao.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from app.dao.service_permissions_dao import (
+    dao_fetch_service_permissions, dao_remove_service_permission)
+from app.models import (
+    EMAIL_TYPE, SMS_TYPE, LETTER_TYPE, INTERNATIONAL_SMS_TYPE, INCOMING_SMS_TYPE, SERVICE_PERMISSION_TYPES)
+
+from tests.app.db import create_service_permission, create_service
+
+
+def test_create_service_permission(sample_service):
+    service_permission_type = SMS_TYPE
+
+    service_permission = create_service_permission(
+        service_id=sample_service.id, permission=service_permission_type)
+
+    assert len(service_permission) == 1
+    assert all(sp.service_id == sample_service.id for sp in service_permission)
+    assert all(sp.permission in service_permission_type for sp in service_permission)
+
+
+def test_fetch_service_permissions_gets_service_permissions(sample_service):
+    service_permission_types = [LETTER_TYPE, EMAIL_TYPE, SMS_TYPE]
+    for spt in service_permission_types:
+        create_service_permission(service_id=sample_service.id, permission=spt)
+    service_permissions = dao_fetch_service_permissions(sample_service.id)
```

Comment:
Generally better to avoid for loops in test as its less explicit. Think here it'd be more readable/debuggable to do:

```
create_service_permission(service_id=sample_service.id, permission=LETTER_TYPE)
create_service_permission(service_id=sample_service.id, permission=EMAIL_TYPE)
create_service_permission(service_id=sample_service.id, permission=SMS_TYPE)
```